vars/dockerExecuteOnKubernetes.groovy

```diff
@@ -171,13 +186,32 @@ void executeOnPod(Map config, utils, Closure body) {
     }
 }
 
+private String generatePodSpec(Map config) {
+    def containers = getContainerList(config)
+    def podSpec = [
+        apiVersion: "v1",
+        kind: "Pod",
+        metadata: [
+            lables: config.uniqueId
+        ],
+        spec: [
+            containers: containers
+        ]
+    ]
+    podSpec.spec.securityContext = getSecurityContext(config)
+
+    return new JsonBuilder(podSpec).toPrettyString()
```

Comment:
Why JsonBuilder for creating yaml?

Reply:
Well, every json document is a yaml document, right?
Yaml is more human readable than json, but since this string is not meant to be read by humans we can also keep it simple and just use json.

Reply:
OK, we then need to make sure that the functionality is serializable.
We should put the JsonBuilder functionality into a function annotated with `@NonCPS`

For Json building there is already one available in another PR: https://github.com/SAP/jenkins-library/pull/548/files#diff-3b30a318e1e23768f29249162c6dc66f

Reply:
Can do this. Is there a way to test if code is CPS compliant?